src/code.h

```diff
@@ -661,12 +683,24 @@ void CodeFuncCallEnd(UInt funccall, UInt options, UInt nr);
 **
 **  'CodeFuncExprEnd'  is an action to  code  a function  expression.  It  is
 **  called when the reader encounters the end of a function expression.  <nr>
-**  is the number of statements in the body of the function.
+**  is the number of statements in the body of the function. If <useStack>
+**  is set, it will not try to make additional returns etc. Also, if an
+**  expression is created (i.e., if the function is not the outermost function)
+**  <useStack> causes the expression to not be put on the stack.
```

Comment:
This paragraph needs to be rewritten, it doesn't really make it clear which values for `useStack` do / mean what. Suggested wording and formatting:
```
...
**  is the number of statements in the body of the function.
**
**  If <useStack> is set to a zero, then the coder's expression stack is not
**  used or modified in any way by this function. In particular, no checks
**  or adjustments for missing trailing return statements are performed.
```

Reply:
I would read it differently tbh. `useStack` for me indicates that the stack is used "as is", i.e., no modification is performed. Maybe we should call it `checkStack` or `modStack` then?

Reply:
I can't really agree with this interpretation in the sense that I am fully convinced many people would be super confused by this use of `useStack`; to me it clearly reads as "use stack!" i.e., an order to use the information on the stack, and possibly also modify it.
However, looking at the code closer, I now see I misunderstood it anyway: I was under the impression that with this flag,  you are not using the stack *at all* (which is why I suggested `useStack` as a name in the first place). But now I see that actually, there are still `PopStat` calls in there (and this is why you are exporting PushStat, to get access to those). Honestly, I am not exactly thrilled about that anyway.

But that makes me wonder why we turn off the return handling logic at all then: this only makes sense if we assume that our input is definitely valid. But we don't know that. So we actually *should* add return statements if necessary, and we should indeed spill extra statements if necessary, etc. So I wouldn't disable this logic at all. I'd only disable the final `PushExpr`....

Reply:
Actually, I already removed the export of `PushExpr` again, only using `NewStatWithProf`.

The main problem is that the checking code in `CodeFuncExprEnd` assumes that the return expr is the last one on the stack. The way we code it though it must not be the last one on the stack, this is why it might not work. So yes, we assume our input is valid (I would think this is a valid assumption atm), and then disable it. I would actually vote for `checkStack` in our context.